### ISSUE TITLE
False positive for `if_same_then_else`

### ISSUE THREAD

**Issue by Skeletonxf** [rust-lang/rust-clippy#5698]
I'm getting what I think is a false positive for the `if_same_then_else` lint

For this function
```rust
pub fn euclidean_length(&self) -> T {
    if self.columns() == 1 {
        // column vector
        (self.transpose() * self).scalar().sqrt()
    } else if self.rows() == 1 {
        // row vector
        (self * self.transpose()).scalar().sqrt()
    } else {
        panic!("Cannot compute unit vector of a non vector, rows: {}, columns: {}",
            self.rows(), self.columns());
    }
}
```

clippy returns an error
```
error: this `if` has identical blocks
   --> src/matrices/mod.rs:830:36
    |
830 |           } else if self.rows() == 1 {
    |  ____________________________________^
831 | |             // row vector
832 | |             (self * self.transpose()).scalar().sqrt()
833 | |         } else {
    | |_________^
    |
    = note: `#[deny(clippy::if_same_then_else)]` on by default
note: same as this
   --> src/matrices/mod.rs:827:32
    |
827 |           if self.columns() == 1 {
    |  ________________________________^
828 | |             // column vector
829 | |             (self.transpose() * self).scalar().sqrt()
830 | |         } else if self.rows() == 1 {
    | |_________^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#if_same_then_else
```

These two blocks are not the same, the order of the matrices on each side of the `*` make the difference between computing the inner product or the outer product.
